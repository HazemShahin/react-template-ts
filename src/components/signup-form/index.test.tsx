import * as React from 'react';
import { shallow } from 'enzyme';

import { SignupForm } from '.';

jest.mock('react-redux', () => ({
	connect: (): jest.Mock => jest.fn()
}));

jest.mock('@src/utilities/hooks', () => ({
	useAppSelector: jest.fn(() => ({
		loading: false,
		loginError: ''
	}))
}));

describe('SignupForm component', () => {
	it('should render successfully', () => {
		const tree = shallow(<SignupForm onSubmit={(): jest.Mock => jest.fn()} />);

		expect(tree).toMatchSnapshot();
	});
});
