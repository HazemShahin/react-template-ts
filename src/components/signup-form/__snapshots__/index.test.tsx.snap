// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm component should render successfully 1`] = `
<form
  className="c-form c-form--signup"
  onSubmit={[Function]}
>
  <h2>
    Signup
  </h2>
  <div
    className="c-form__cols"
  >
    <div
      className="c-form__col"
    >
      <Component
        label="Email Address"
        name="email"
        placeholder="someone@example.com"
        register={[Function]}
        type="email"
      />
      <Component
        label="Password"
        name="password"
        placeholder="********"
        register={[Function]}
        type="password"
      />
      <Component
        label="Confirm password"
        name="confirm"
        placeholder="********"
        register={[Function]}
        type="password"
      />
      <Component
        label="Username"
        name="username"
        placeholder="ShortWhale"
        register={[Function]}
        type="text"
      />
    </div>
    <div
      className="c-form__col"
    >
      <Component
        label="First name"
        name="first_name"
        placeholder="John"
        register={[Function]}
        type="text"
      />
      <Component
        label="Last name"
        name="last_name"
        placeholder="Smith"
        register={[Function]}
        type="text"
      />
      <Component
        label="Gender"
        name="gender"
        options={
          Array [
            "Unspecified",
            "Male",
            "Female",
          ]
        }
        register={[Function]}
        type="select"
      />
      <Component
        label="Avatar image URL"
        name="image_url"
        placeholder="https://images.com/john-smith.jpg"
        register={[Function]}
        type="url"
      />
    </div>
  </div>
  <Component
    className=""
    disabled={false}
    type="submit"
  >
    Sign up
  </Component>
</form>
`;
